quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -90,6 +103,31 @@ public class PropertyMapper<T> {
         this.validator = validator;
         this.description = description;
         this.parentMapper = parentMapper;
+
+
+        // The wildcard pattern (e.g. log-level-<category>) is matching only a-z, 0-0 and dots. For env vars, dots are replaced by underscores.
+        if (option.getKey() != null) {
+            Matcher matcher = WILDCARD_PLACEHOLDER_PATTERN.matcher(option.getKey());
+            if (matcher.find()) {
+                // Includes handling for both "--" prefix for CLI options and "kc." prefix
+                this.optionNameWildcardPattern = Pattern.compile("(?:" + ARG_PREFIX + "|kc\\.)" + matcher.replaceFirst("([\\\\\\\\.a-zA-Z0-9]+)"));
+
+                // Not using toEnvVarFormat because it would process the whole string incl the <...> wildcard.
+                Matcher envVarMatcher = WILDCARD_PLACEHOLDER_PATTERN.matcher(option.getKey().toUpperCase().replace("-", "_"));
+                this.envVarNameWildcardPattern = Pattern.compile("KC_" + envVarMatcher.replaceFirst("([_A-Z0-9]+)"));
+
+                if (to != null) {
+                    toWildcardMatcher = WILDCARD_PLACEHOLDER_PATTERN.matcher(to);
+                    if (!toWildcardMatcher.find()) {
+                        throw new IllegalArgumentException("Attempted to map a wildcard option to a non-wildcard option");
+                    }
+
+                    this.toWildcardPattern = Pattern.compile(toWildcardMatcher.replaceFirst("([\\\\\\\\.a-zA-Z0-9]+)"));
+                }
+            }
+
+            this.wildcardValuesTransformer = wildcardValuesTransformer;
+        }
```

Comment:
@vmuzikar From the first sight, it looks good! But I didn't have time to look at it closer, but I'm curious if SmallRye Config supports something similar that could be used. As Quarkus supports specifying the log categories or multiple named datasources, they need to have some sort of functionality already implemented. 

Did you have a chance to analyze it?